src/Service.Host/client/src/components/saHoldStories/SaHoldStory.js

```diff
@@ -17,54 +17,60 @@ function SaHoldStory({
     errorMessage,
     editStatus,
     item,
+    itemId,
     addSaHoldStory,
     setEditStatus,
     snackbarVisible,
     setSnackbarVisible,
     history,
-    match
+    match,
+    updateSaHoldStory
 }) {
     const deslugify = articleNumber => articleNumber.replace(/%2F/g, '/');
     const slugify = articleNumber => articleNumber.replace(/[/]/g, '%2F');
 
     const creating = () => editStatus === 'create';
     const viewing = () => editStatus === 'view';
+    const editing = () => editStatus === 'edit';
 
-    const [saHoldStory, setSaHoldStory] = useState(
-        !creating()
-            ? {}
-            : {
-                  salesArticle: deslugify(match.params.articleNumber),
-                  dateStarted: moment().toISOString()
-              }
-    );
+    const initialState = () => {
+        if (creating()) {
+            return {
+                salesArticle: deslugify(match.params.articleNumber),
+                dateStarted: moment().toISOString()
+            };
+        }
+        return {};
+    };
+
+    const [saHoldStory, setSaHoldStory] = useState(initialState);
     const [prevSaHoldStory, setPrevSaHoldStory] = useState({});
 
     useEffect(() => {
         if (!creating() && item !== prevSaHoldStory) {
-            setSaHoldStory(item);
+            setSaHoldStory(editing() ? { ...item, dateFinished: moment().toISOString() } : item);
             setPrevSaHoldStory(item);
```

Comment:
if editing is false then both the current and previous are the same item.. not saying that's wrong.. would use effect only be invoked whilst editing?